diff -ur ghostscript-10.03.1.old/devices/gdevescp.c ghostscript-10.03.1/devices/gdevescp.c
--- ghostscript-10.03.1.old/devices/gdevescp.c	2024-05-02 11:45:25.000000000 +0200
+++ ghostscript-10.03.1/devices/gdevescp.c	2024-07-10 10:47:10.751123360 +0200
@@ -162,7 +162,7 @@
         ** margin measured from the *top* of the page:
         */
 
-        fwrite("\033(U\001\0\n\033(C\002\0t\020\033(c\004\0\0\0t\020",
+        gp_fwrite("\033(U\001\0\n\033(C\002\0t\020\033(c\004\0\0\0t\020",
                                                         1, 22, prn_stream);
 #endif
 
