--- gs/lib/ps2epsi	Tue Sep 19 20:29:11 2000
+++ gs/lib/ps2epsi	Wed Jan 10 21:16:31 2001
@@ -51,7 +51,7 @@
 		}
 	' U="$USERNAME$LOGNAME"  F=1 - F=2 ${infile} >$tmpfile
 
-gs -q -dNOPAUSE -dSAFER -r72 -sDEVICE=bit -sOutputFile=/dev/null $tmpfile ps2epsi.ps $tmpfile <${infile} 1>&2
+gs -q -dNOPAUSE -r72 -sDEVICE=bit -sOutputFile=/dev/null $tmpfile ps2epsi.ps $tmpfile <${infile} 1>&2
 rm -f $tmpfile
 
 (
@@ -62,10 +62,8 @@
 BEGINEPS
 
 cat ${infile} |
-sed -e '/^%%BeginPreview:/,/^%%EndPreview[
-]*$/d' -e '/^%!PS-Adobe/d'\
-	-e '/^%%[A-Za-z][A-Za-z]*[
-]*$/d' -e '/^%%[A-Za-z][A-Za-z]*: /d'
+sed -e '/^%%BeginPreview:/,/^%%EndPreview[\n]*$/d' -e '/^%!PS-Adobe/d'\
+	-e '/^%%[A-Za-z][A-Za-z]*[\n]*$/d' -e '/^%%[A-Za-z][A-Za-z]*: /d'
 
 cat << ENDEPS
 %%Trailer
--- gs/lib/pdf2dsc.orig	Wed Jan 10 21:25:24 2001
+++ gs/lib/pdf2dsc	Wed Jan 10 21:25:32 2001
@@ -22,5 +22,5 @@
 dscfile=$2
 : ${dscfile:=`echo $pdffile | sed 's,\.[^/.]*,,'`.dsc}
 
-exec gs -q -dNODISPLAY -dSAFER \
+exec gs -q -dNODISPLAY \
     -sPDFname="$pdffile" -sDSCname="$dscfile" pdf2dsc.ps -c quit
