--- espgs-7.07.1/configure.ac.orig	2003-11-22 00:38:50.000000000 +0100
+++ espgs-7.07.1/configure.ac	2003-11-22 00:41:27.000000000 +0100
@@ -531,14 +531,7 @@
     dnl
     dnl Test for glib package.
     dnl
-    AC_PATH_PROG(GLIB_CONFIG, glib-config)
-    AC_MSG_CHECKING([for glib])
-    if test ! -z "$GLIB_CONFIG"; then
-        AC_MSG_RESULT([yes])
-    else
-        AC_MSG_RESULT([no])
-        INCLUDEOMNI=no
-    fi
+    PKG_CHECK_MODULES(GLIB, gmodule-2.0, , INCLUDEOMNI=no)
 
     dnl
     dnl If all tests pass, then include omni support!
@@ -548,8 +541,8 @@
         AC_MSG_RESULT([yes])
 
         OMNIDEVS='$(DD)omni.dev'
-        CFLAGS="$CFLAGS `$GLIB_CONFIG --cflags gmodule`"
-        LIBS="$LIBS `$GLIB_CONFIG --libs gmodule`"
+        CFLAGS="$CFLAGS $GLIB_CFLAGS"
+        LIBS="$LIBS $GLIB_LIBS"
 
         if test -n "$GCC"; then
 	    LIBS="$LIBS -lstdc++"
