--- ghostscript-8.51/configure.ac.am	2005-01-25 02:19:34.000000000 +0100
+++ ghostscript-8.51/configure.ac	2005-07-19 22:12:23.000000000 +0200
@@ -147,6 +147,32 @@
 	GCFLAGS="$GCFLAGS -DGX_COLOR_INDEX_TYPE=\"$uint64_type\""
 fi
 
+dnl look for default font path...
+AC_ARG_WITH(fontpath, [  --with-fontpath         set font path for gs],fontpath="$withval",fontpath="")
+
+dnl Fix "prefix" variable...
+if test "x$prefix" = xNONE; then
+        prefix=/usr/local
+fi
+
+dnl Fix "datadir" variable...
+if test "x$datadir" = 'x${prefix}/share'; then
+        datadir="$prefix/share"
+fi
+	dnl Fix "fontpath" variable...
+if test "x$fontpath" = "x"; then
+# These font directories are used by various Linux distributions...
+     fontpath="$datadir/fonts/default/ghostscript"
+     fontpath="${fontpath}:$datadir/fonts/default/Type1"
+     fontpath="${fontpath}:$datadir/fonts/default/TrueType"
+     # These font directories are used by IRIX...
+     fontpath="${fontpath}:/usr/lib/DPS/outline/base"
+     # These font directories are used by Solaris...
+     fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/Type1"
+     fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/TrueType"
+fi
+
+AC_SUBST(fontpath)
 
 dnl --------------------------------------------------
 dnl Check for libraries
--- ghostscript-8.51/src/Makefile.in~	2004-12-01 09:17:19.000000000 +0100
+++ ghostscript-8.51/src/Makefile.in	2005-07-21 13:46:26.000000000 +0200
@@ -70,7 +70,7 @@
 # Define the default directory/ies for the runtime
 # initialization and font files.  Separate multiple directories with a :.
 
-GS_LIB_DEFAULT=$(gsdatadir)/lib:$(gsdatadir)/Resource:$(gsdir)/fonts
+GS_LIB_DEFAULT=$(gsdatadir)/lib:$(gsdatadir)/Resource:$(gsdir)/fonts:@fontpath@
 
 # Define the default directory for cached data files
 # this must be a single path.
