--- ghostscript-9.54.0/configure.ac.orig	2021-04-25 20:45:17.498439464 +0200
+++ ghostscript-9.54.0/configure.ac	2021-04-25 20:46:23.098488486 +0200
@@ -1403,7 +1403,7 @@ AC_MSG_CHECKING([for local zlib source])
 dnl zlib is needed for language level 3, and libpng
 # we must define ZLIBDIR regardless because png.mak does a -I$(ZLIBDIR)
 # this seems a harmless default
-ZLIBDIR=src
+ZLIBDIR=$includedir
 AUX_SHARED_ZLIB=
 ZLIBCFLAGS=""
 
@@ -1434,6 +1434,8 @@ if test -z "$SHARE_ZLIB"; then
   local directory named 'zlib'. See http://www.gzip.org/zlib/
   for more information.
   ])
+else
+  LIBS="$LIBS -lz"
 fi
 
 dnl if GS is to use the system zlib, freetype
