--- ghostscript-8.51/configure.ac.am	2005-01-25 02:19:34.000000000 +0100
+++ ghostscript-8.51/configure.ac	2005-07-19 22:12:23.000000000 +0200
@@ -147,6 +147,32 @@
 	GCFLAGS="$GCFLAGS -DGX_COLOR_INDEX_TYPE=\"$uint64_type\""
 fi
 
+dnl look for default font path...
+AC_ARG_WITH(fontpath, [  --with-fontpath         set font path for gs],fontpath="$withval",fontpath="")
+
+dnl Fix "prefix" variable...
+if test "x$prefix" = xNONE; then
+        prefix=/usr/local
+fi
+
+dnl Fix "datadir" variable...
+if test "x$datadir" = 'x${prefix}/share'; then
+        datadir="$prefix/share"
+fi
+	dnl Fix "fontpath" variable...
+if test "x$fontpath" = "x"; then
+# These font directories are used by various Linux distributions...
+     fontpath="$datadir/fonts/default/ghostscript"
+     fontpath="${fontpath}:$datadir/fonts/default/Type1"
+     fontpath="${fontpath}:$datadir/fonts/default/TrueType"
+     # These font directories are used by IRIX...
+     fontpath="${fontpath}:/usr/lib/DPS/outline/base"
+     # These font directories are used by Solaris...
+     fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/Type1"
+     fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/TrueType"
+fi
+
+AC_SUBST(fontpath)
 
 dnl --------------------------------------------------
 dnl Check for libraries
@@ -306,55 +332,13 @@
 SHARE_JASPER=0
 JPXDEVS=''
 if test x$with_jasper != xno; then
-  AC_MSG_CHECKING([for local jasper library source])
-  for d in jasper jasper-1.7*; do
-    test -d "$d" && JASPERDIR=$d && break
-  done
-  if test "x$JASPERDIR" != xsrc; then
-    AC_MSG_RESULT([$JASPERDIR])
-    AC_MSG_CHECKING([for local jasper configure script])
-    if test -x $JASPERDIR/configure; then
-      AC_MSG_RESULT([yes])
-      echo
-      echo "Running jasper configure script..."
-      olddir=`pwd`
-      cd $JASPERDIR && ./configure
-      cd $olddir
-      echo
-      echo "Continuing with Ghostscript configuration..."
-    else
-      AC_MSG_RESULT([no])
-      AC_MSG_CHECKING([for local jasper autogen.sh script])
-      if test -x $JASPERDIR/autogen.sh; then
-        AC_MSG_RESULT([yes])
-        echo
-        echo "Running jasper autogen script..."
-        olddir=`pwd`
-        cd $JASPERDIR && ./autogen.sh
-        cd $olddir
-        echo
-        echo "Continuing with Ghostscript configuration..."
-      else
-        AC_MSG_ERROR([
-Unable to find $JASPERDIR/src/libjasper/include/jas_config.h,
-or generate generate such a file for this system. You will
-have to build one by hand, or configure, build and install
-the jasper library separately.
-
-You can also pass --without-jasper to configure to disable
-JPEG 2000 PDF image support entirely.
-])
-      fi
-    fi
-  else
-    AC_MSG_RESULT([no])
+  JASPERDIR=
     AC_CHECK_LIB([jasper], [jas_image_create], [
 	SHARE_JASPER=1
     ], [
 	AC_MSG_WARN([disabling support for JPEG 2000 images])
         with_jasper=no
     ])
-  fi
 fi
 if test x$with_jasper != xno; then
   JPXDEVS='$(PSD)jpx.dev'
--- ghostscript-8.51/src/Makefile.in~	2004-12-01 09:17:19.000000000 +0100
+++ ghostscript-8.51/src/Makefile.in	2005-07-21 13:46:26.000000000 +0200
@@ -70,7 +70,7 @@
 # Define the default directory/ies for the runtime
 # initialization and font files.  Separate multiple directories with a :.
 
-GS_LIB_DEFAULT=$(gsdatadir)/lib:$(gsdatadir)/Resource:$(gsdir)/fonts
+GS_LIB_DEFAULT=$(gsdatadir)/lib:$(gsdatadir)/Resource:$(gsdir)/fonts:@fontpath@
 
 # Define the default directory for cached data files
 # this must be a single path.
