diff -Nur src.orig/contrib.mak src/contrib.mak
--- src.orig/contrib.mak	Tue Mar 27 23:21:02 2001
+++ src/contrib.mak	Tue Mar 27 23:26:05 2001
@@ -553,6 +553,36 @@
 $(DD)pjetxl.dev : $(PJET) $(DD)page.dev
 	$(SETPDEV) $(DD)pjetxl $(PJET)
 
+
+### ------------- Hewlett-Packard Co. Inkjet Driver -------------- ###
+
+hpijs_=$(GLOBJ)gdevhpij.$(OBJ)
+
+$(GLOBJ)gdevhpij.$(OBJ): $(GLSRC)gdevhpij.c $(PDEVH)
+	$(GLCC) $(GLO_)gdevhpij.$(OBJ) $(C_) $(GLSRC)gdevhpij.c
+
+$(DD)DJ630.dev : $(hpijs_) $(DD)page.dev
+	$(SETPDEV) $(DD)DJ630 $(hpijs_)
+
+$(DD)DJ6xx.dev : $(hpijs_) $(DD)page.dev
+	$(SETPDEV) $(DD)DJ6xx $(hpijs_)
+
+$(DD)DJ6xxP.dev : $(hpijs_) $(DD)page.dev
+	$(SETPDEV) $(DD)DJ6xxP $(hpijs_)
+
+$(DD)DJ8xx.dev : $(hpijs_) $(DD)page.dev
+	$(SETPDEV) $(DD)DJ8xx $(hpijs_)
+
+$(DD)DJ9xx.dev : $(hpijs_) $(DD)page.dev
+	$(SETPDEV) $(DD)DJ9xx $(hpijs_)
+
+$(DD)DJ9xxVIP.dev : $(hpijs_) $(DD)page.dev
+	$(SETPDEV) $(DD)DJ9xxVIP $(hpijs_)
+
+$(DD)AP21xx.dev : $(hpijs_) $(DD)page.dev
+	$(SETPDEV) $(DD)AP21xx $(hpijs_)
+
+								
 ###--------------------- The Brother HL 7x0 printer --------------------- ### 
 ### Note: this driver was contributed by users: please contact            ###
 ###       Pierre-Olivier Gaillard (pierre.gaillard@hol.fr)                ###
diff -Nur src.orig/unix-gcc.mak src/unix-gcc.mak
--- src.orig/unix-gcc.mak	Tue Mar 27 23:21:02 2001
+++ src/unix-gcc.mak	Tue Mar 27 23:39:33 2001
@@ -40,7 +40,9 @@
 #include $(COMMONDIR)/gccdefs.mak
 #include $(COMMONDIR)/unixdefs.mak
 #include $(COMMONDIR)/generic.mak
-include $(GLSRCDIR)/version.mak
+GLSRC=$(GLSRCDIR)/
+include $(GLSRC)version.mak
+PSSRC=$(PSSRCDIR)/
 DD=$(GLGENDIR)/
 GLD=$(GLGENDIR)/
 PSD=$(PSGENDIR)/
@@ -355,7 +357,7 @@
 DEVICE_DEVS1=$(DD)bmpmono.dev $(DD)bmpgray.dev $(DD)bmpsep1.dev $(DD)bmpsep8.dev $(DD)bmp16.dev $(DD)bmp256.dev $(DD)bmp16m.dev $(DD)bmp32b.dev
 DEVICE_DEVS2=$(DD)bmpamono.dev $(DD)bmpasep1.dev $(DD)bmpasep8.dev $(DD)bmpa16.dev $(DD)bmpa256.dev $(DD)bmpa16m.dev $(DD)bmpa32b.dev
 DEVICE_DEVS3=$(DD)deskjet.dev $(DD)djet500.dev $(DD)laserjet.dev $(DD)ljetplus.dev $(DD)ljet2p.dev $(DD)ljet3.dev $(DD)ljet3d.dev $(DD)ljet4.dev $(DD)ljet4d.dev $(DD)lj5mono.dev $(DD)lj5gray.dev
-DEVICE_DEVS4=$(DD)cdeskjet.dev $(DD)cdjcolor.dev $(DD)cdjmono.dev $(DD)cdj550.dev $(DD)pj.dev $(DD)pjxl.dev $(DD)pjxl300.dev
+DEVICE_DEVS4=$(DD)cdeskjet.dev $(DD)cdjcolor.dev $(DD)cdjmono.dev $(DD)cdj550.dev $(DD)pj.dev $(DD)pjxl.dev $(DD)pjxl300.dev $(DD)DJ630.dev $(DD)DJ6xx.dev $(DD)DJ6xxP.dev $(DD)DJ8xx.dev $(DD)DJ9xx.dev $(DD)DJ9xxVIP.dev $(DD)AP21xx.dev
 DEVICE_DEVS5=$(DD)uniprint.dev
 DEVICE_DEVS6=$(DD)bj10e.dev $(DD)bj200.dev $(DD)bjc600.dev $(DD)bjc800.dev
 DEVICE_DEVS7=$(DD)faxg3.dev $(DD)faxg32d.dev $(DD)faxg4.dev
