--- ghostscript-9.02/base/configure.ac~	2011-06-11 11:51:32.885645350 +0200
+++ ghostscript-9.02/base/configure.ac	2011-06-11 11:52:19.970579073 +0200
@@ -909,7 +909,7 @@
                 LCUPSINCLUDE="include \$(GLSRCDIR)/lcups.mak"
                 LCUPSIINCLUDE="include \$(GLSRCDIR)/lcupsi.mak"
                 if ( test x$with_pdftoraster != xno ); then
-                    if test "$CUPSVERSION" ">" "1.2"; then
+                    if echo "$CUPSVERSION" | grep -q '^1\.[[2-9]]'; then
                         CUPSPDFTORASTER="1"
                     fi
                 fi

