--- espgs-7.05.6/ijs/Makefile.in.orig	2003-07-09 23:08:13.000000000 +0200
+++ espgs-7.05.6/ijs/Makefile.in	2003-07-09 23:50:25.000000000 +0200
@@ -1,7 +1,7 @@
 # Unix
 
 CC=@CC@
-CFLAGS=-g -Wall -ansi -pedantic -Wmissing-prototypes
+CFLAGS=@CFLAGS@ -Wall -ansi -pedantic -Wmissing-prototypes
 LDLIBS=
 OBJ=.@OBJEXT@
 EXE=@EXEEXT@
@@ -36,6 +36,11 @@
 
 LIB_OBJS=ijs$(OBJ) ijs_client$(OBJ) ijs_server$(OBJ) $(IJS_EXEC_SERVER)
 
+SHLIB_OBJS=ijs.pic$(OBJ) ijs_client.pic$(OBJ) ijs_server.pic$(OBJ) ijs_exec_unix.pic$(OBJ)
+
+%.pic$(OBJ) : %.c
+	$(CC) $(CFLAGS) -fPIC -c -o $@ $<
+
 libijs.a:	$(LIB_OBJS)
 	rm -f $@
 	$(AR) $(ARFLAGS) $@ $^
@@ -44,8 +49,8 @@
 # Note: this builds both the server and client into a single library. Logically, it
 # makes sense to separate them, but they're small enough to make this probably
 # not worthwhile.
-libijs$(SHARED_OBJ):	$(LIB_OBJS)
-	$(CC) $(SHARED_LDFLAGS) $^ -o $@
+libijs$(SHARED_OBJ):	$(SHLIB_OBJS)
+	$(CC) $(SHARED_LDFLAGS) $^ -o $@ -Wl,-soname=$@
 
 ijs_client_example$(EXE):	ijs_client_example$(OBJ) ijs_client$(OBJ) $(IJS_COMMON_OBJ) $(IJS_EXEC_SERVER)
 	$(CC) $(CFLAGS) $(FE)ijs_client_example$(EXE) ijs_client_example$(OBJ) ijs_client$(OBJ) $(IJS_COMMON_OBJ) $(IJS_EXEC_SERVER) $(LDLIBS)
